src/Contract/Config/CollectorConfig.php

```diff
@@ -16,7 +16,7 @@ public function private(): self
         return $this;
     }
 
-    /** @return array{'type': string, 'private': bool} */
+    /** @return array{'type': string, 'private': bool, ...} */
```

Comment:
This is the cause of your `psalm` issues. I would recommend rolling it back and baselining the issue in `TagValueRegexConfig`.

Reply:
I would like to understand this better. The way this is declared right now, it means that the only two fields that toArray() is allowed to return are "type" and "private". But most subclasses do return additional fields. Why should we not explicitly allow that by marking the shape as "unsealed"?

The error I get on github is actually what I'd expect - unsealing the shape resolves the baseline issues, and they should be deleted from the baseline file. What confuses me is that psalm doesn't tell me that when I run it locally...

Reply:
No, you have it correct. I just did not want to mix the solution with the PR that is about something else. But since you have opened the can of worms, you might as well can. 

I think the reason you don't get the `psalm` to complain locally is cache. Since you have not changed those other files, I think psalm does not analyze them for issues in those files. But in CI there is no cache. It does a full analysis and therefore discovers those issues. You might try to run psalm locally with `--no-cache` or `--clear-cache` and re-run it again.

Reply:
done